### DISCUSSION TITLE
index wb cache problem for merge

### DISCUSSION THREAD

**Discussion by shosseinimotlagh** [eBay/HomeStore#513] [General]
index wb cache problem for merge

**Comment by shosseinimotlagh** [comment]
[https://lucid.app/lucidchart/b49a93b5-863b-4c69-b108-1cea20fc9326/edit?invitationId=inv_2a4d0928-083d-409a-82f9-5eba8dcd4a81&page=Wx8WcXXTmdt8#](https://lucid.app/lucidchart/b49a93b5-863b-4c69-b108-1cea20fc9326/edit?invitationId=inv_2a4d0928-083d-409a-82f9-5eba8dcd4a81&page=Wx8WcXXTmdt8#)

**Reply by shosseinimotlagh** [reply]
@jkou@ebay.com

Graph for the current basic merge test case:
https://lucid.app/lucidchart/b49a93b5-863b-4c69-b108-1cea20fc9326/edit?invitationId=inv_2a4d0928-083d-409a-82f9-5eba8dcd4a81&page=Wx8WcXXTmdt8#
I think we’ve talked about including freed nodes in the DAG before, but now it seems that node-freeing during the merge process actually depends on the persistence of the leftmost node - old nodes can be freed only when the leftmost node is flushed, otherwise the entry distribution would be messed up.
So it’s a one-to-many dependency rather than a simple up-down tree structure for the merge case.

**Reply by shosseinimotlagh** [reply]
Jilong made a simple/temp approach of including freed nodes in the dependency tree [here](https://github.com/koujl/HomeStore/commit/a0a3977bc5d84487285a861e11a20e19dc150a86), and it can just pass the basic merge UT. But seems for some more complicated cases this would become a lot more complex to maintain/recover the DAG correctly…

**Reply by shosseinimotlagh** [reply]
adding a separate deleted-nodes mgmt mechanism (like a leftmost-freed buf mapping) in cp flushing for such one-to-many dependency, but this will surely involve many code refactoring in cp flushing & recovering logic -

**Reply by hkadayam** [reply]
Given that Lucidchart is internal, I am unable to understand the actual problem.
